server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -362,10 +373,8 @@ private boolean syncToDisk() throws IOException {
     @Override
     public void sync() throws IOException {
         try {
-            if (syncToDisk() || syncNeeded()) {
-                if (prepareForUpload(null)) {
-                    performUpload(primaryTermSupplier.getAsLong(), null);
-                }
+            if (syncToDisk() || syncNeeded() || prepareForUpload(null)) {
```

Comment:
shouldn't it be `if ( (syncToDisk() || syncNeeded()) && prepareForUpload(null))`